staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go

```diff
@@ -1095,9 +1095,12 @@ type ManagedFieldsEntry struct {
 	// Time is timestamp of when these fields were set. It should always be empty if Operation is 'Apply'
 	// +optional
 	Time *Time `json:"time,omitempty" protobuf:"bytes,4,opt,name=time"`
-	// Fields identifies a set of fields.
+
+	// +k8s:deprecated=fields,protobuf=5
+
+	// FieldSet identifies a set of fields.
 	// +optional
-	Fields *Fields `json:"fields,omitempty" protobuf:"bytes,5,opt,name=fields,casttype=Fields"`
+	FieldSet *runtime.RawExtension `json:"fieldSet,omitempty" protobuf:"bytes,6,opt,name=fieldSet,casttype=k8s.io/apimachinery/pkg/runtime.RawExtension"`
```

Comment:
We talked about e.g. versioning the contents of this field, since we won't be permitted to break it like this again once we're beta. Any thoughts about how we might do that?

Is there a reason why you're not going with my trick of keeping the field name but changing the proto number?

Reply:
> We talked about e.g. versioning the contents of this field, since we won't be permitted to break it like this again once we're beta. Any thoughts about how we might do that?

Versioning could be done per managed fields entry, but we might also want to consider versioning the entire list together.


> Is there a reason why you're not going with my trick of keeping the field name but changing the proto number?

To me it seemed easier to reason about removing a field and adding a new field, than changing the proto number and not the field name.